luaui/Widgets/gui_pregameui.lua

```diff
@@ -79,10 +81,267 @@ local buttonRect = {0,0,0,0}
+    end
+end
+local function DrawRankImage(rankImage, posX, posY)
+    gl_Color(1, 1, 1, 1)
+    gl_Texture(rankImage)
+    DrawRect(posX + (3*playerScale), posY + (8*playerScale) - (7.5*playerScale), posX + (17*playerScale), posY + (8*playerScale) + (7.5*playerScale))
+end
+local function DrawRank(rank, posX, posY)
+    if rank == 0 then
+        DrawRankImage(pics["rank0"],  posX, posY)
+    elseif rank == 1 then
+        DrawRankImage(pics["rank1"],  posX, posY)
+    elseif rank == 2 then
+        DrawRankImage(pics["rank2"],  posX, posY)
+    elseif rank == 3 then
+        DrawRankImage(pics["rank3"],  posX, posY)
+    elseif rank == 4 then
+        DrawRankImage(pics["rank4"],  posX, posY)
+    elseif rank == 5 then
+        DrawRankImage(pics["rank5"],  posX, posY)
+    elseif rank == 6 then
+        DrawRankImage(pics["rank6"],  posX, posY)
+    elseif rank == 7 then
+        DrawRankImage(pics["rank7"],  posX, posY)
+    else
+
+    end
+end
+local function DrawSkill(skill, posX, posY)
+    local fontsize = 9.5 * (playerScale + ((1-playerScale)*0.25))
+    font:Begin()
+    font:Print(skill, posX + (4.5*playerScale), posY + (5.3*playerScale), fontsize, "o")
+    font:End()
+end
+-- advplayerlist end
+local function colourNames(teamID)
+	if teamID == nil then return "\255\210\210\210" end -- debug
+	local nameColourR, nameColourG, nameColourB, nameColourA = Spring.GetTeamColor(teamID)
+	local R255 = math.floor(nameColourR * 255)
+	local G255 = math.floor(nameColourG * 255)
+	local B255 = math.floor(nameColourB * 255)
+	if R255 % 10 == 0 then
+		R255 = R255 + 1
+	end
+	if G255 % 10 == 0 then
+		G255 = G255 + 1
+	end
+	if B255 % 10 == 0 then
+		B255 = B255 + 1
+	end
+	return "\255" .. string.char(R255) .. string.char(G255) .. string.char(B255)
+end
+local function canPlayerPlaceNow(playerID) -- returns true if playerID is found before array hits index (currentPlayerIndex)
+	if draftMode == nil or draftMode == "disabled" or draftMode == "fair" then return true end
+	if currentPlayerIndex == nil or currentPlayerIndex <= 0 or myTeamPlayersOrder == nil then
+		return false
+	end
+	for i = 1, #myTeamPlayersOrder do
+		if (i > currentPlayerIndex) then
+			return false
+		end
+        if myTeamPlayersOrder[i].id == playerID then
+            return true
+        end
+    end
+    return false
+end
+local function findPlayerName(playerID)
+	if myTeamPlayersOrder then
+		for _, player in ipairs(myTeamPlayersOrder) do
+			if player.id == playerID then
+				if player.name ~= nil then
+					return player.name
+				else -- try to cache missing player name
+					local tname = select(1, Spring.GetPlayerInfo(playerID, false))
+					if tname ~= nil then
+						player.name = tname
+						return player.name
+					end
+				end
+			end
+		end
+	end
+	tname = select(1, Spring.GetPlayerInfo(playerID, false)) or "unconnected" -- show "unconnected" instead of nil if we don't know the name
```

Comment:
`tname` needs to be declared as a local var

Reply:
Fixed.